### ISSUE TITLE
Checkstyle problem : Checkstyle execution failed due to an internal error

### ISSUE THREAD

**Issue by sukanta-27** [TEAMMATES/teammates#8314]
**Environment**

OS: Windows 8.1
IDE: Eclipse Oxygen.1a Release (4.7.1a)
Checkstyle Version: 8.7.0 

While working on the PR #8182, I, after doing some changes in `EmailGenerator.java` checked the code with checkstyle and found an error [Screenshot below]. First I thought there is some error in my code, but the `gradlew.bat lint` and `npm run lint` both ran successfully. So, I checked out to the `master` branch to check if this error persists or not. And turns out this error occurs in the master branch also.<br><br> I checked the error log and found this error : 
```
!ENTRY net.sf.eclipsecs.core 4 0 2018-01-19 14:32:15.689
!MESSAGE Checkstyle-Plugin: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
!STACK 0
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:457)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createCheckerInternal(CheckerFactory.java:290)
	at net.sf.eclipsecs.core.builder.CheckerFactory.createChecker(CheckerFactory.java:128)
	at net.sf.eclipsecs.core.builder.Auditor.runAudit(Auditor.java:141)
	at net.sf.eclipsecs.core.builder.CheckstyleBuilder.handleBuildSelection(CheckstyleBuilder.java:301)
	at net.sf.eclipsecs.core.jobs.RunCheckstyleOnFilesJob.runInWorkspace(RunCheckstyleOnFilesJob.java:119)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:39)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: TreeWalker is not allowed as a parent of SuppressionCommentFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.
	at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:155)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:452)
	... 8 more
```
**Screenshot of the error:**
![screenshot 297](https://user-images.githubusercontent.com/10681755/35143384-45d94ba6-fcf9-11e7-8fb8-3292181af165.png)

**Steps to reproduce:**
To reproduce this error right click on any java file and then `Checkstyle -> Check Code with Checkstyle` and this error will appear in the markers. 

While looking for a solution for this error I found this [thread](https://sourceforge.net/p/eclipse-cs/discussion/274376/thread/ba316e08/?limit=25) which addresses this error, but I haven't found a solution for it yet. Can anyone confirm if they are also getting this error or not?

**Comment by wkurniawan07** [comment]
@sukanta-27 one of the following should be the case:
- You're working on a branch that is in sync with the latest `master`, but your Checkstyle version in Eclipse is below `8.5.0`. The Checkstyle version in `build.gradle` is not relevant right now.
- Your Checkstyle version in Eclipse is `8.5.0` or above, but you're working on a branch that is not in sync with the latest `master`, i.e. still using the old Checkstyle config.

**Comment by sukanta-27** [comment]
Below is a gif of the error I am getting:
![output_hsrbkb](https://user-images.githubusercontent.com/10681755/35145278-246fed52-fcff-11e7-93c5-6fccfb129827.gif)

**Comment by wkurniawan07** [comment]
That gif is not helpful. The error is expected when the plugin version and the config file does not match.

**Comment by sukanta-27** [comment]
@wkurniawan07 Thank you very much for the quick response. I was looking into the `build.gradle` and mistakenly thought the version specified in the file was my plug-in version. Sorry for the late response, as whenever I was trying to install the checkstyle plug-in through the eclipse marketplace I ran into this problem : 
```
Unable to read repository at http://eclipse-cs.sourceforge.net/update/compositeContent.xml.
Unable to read repository at http://eclipse-cs.sourceforge.net/update/compositeContent.xml.
eclipse-cs.sourceforge.net:80 failed to respond
```
Though I could not solve this issue, even after trying out many solutions from [StackOverflow](https://stackoverflow.com/questions/7625943/unable-to-read-repository-at-http-download-eclipse-org-releases-indigo) site and other sites, I finally installed the plug-in manually and the issue was resolved.
